### PR TITLE
[gru] enable bias_hh, reset_after

### LINKED ISSUE #1768: Support for two biases for GRU and LSTM
Require support for two biases for GRU and LSTM. 
If the GRU and LSTM are pre-trained and to be finetuned, the effect of two biases cannot be handled with 1 bias. So, our implementations of GRU and LSTM should support two biases.

### PR DESCRIPTION
**Pull request by lhs8928**
- Enable bias_hh in gru.
 - Enable reset_after in gru, grucell. If reset_after is set to true,
   apply reset gate after matrix multiplication.

close #1768

Self evaluation:

Build test: [X]Passed [ ]Failed [ ]Skipped
Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: nntrainer/layers/grucell.cpp [hunk 1]
```diff
@@ -262,13 +263,22 @@ void GRUCellLayer::forwarding(RunLayerContext &context, bool training) {
...
+    prev_hidden_state.dot(weight_hh_g, temp);
+    if (!disable_bias && !integrate_bias) {
+      Tensor bias_hh_g = bias_hh.getSharedDataTensor({unit}, 2 * unit);
+      temp.add_i(bias_hh_g);
+    }
+    temp.multiply_i_strided(r_gate);
+    g_gate.add_i_strided(temp);
+  } else {
+    r_gate.multiply_strided(prev_hidden_state, temp);
+    g_gate.add_i(temp.dot(weight_hh_g));
```
Comment (kparichay) [inline nntrainer/layers/grucell.cpp:276]: minor comment - `temp.dot(weight_hh_g, g_gate, 1.0)`

File: test/input_gen/transLayer.py [hunk 2]
```diff
@@ -213,6 +213,24 @@ def call(self, x, training=None):
...
+        else:
+            return tensorOrList
+
+    def to_nntr_trainable_weights(self, tensorOrList):
+        bias = tensorOrList[2]
+        if bias.shape.rank == 2:
+            bias_ih, bias_hh = bias[0], bias[1]
+            return [tensorOrList[0], tensorOrList[1], bias_ih, bias_hh]
+        else:
+            return tensorOrList
```
Comment (zhoonit) [inline test/input_gen/transLayer.py:233]: nit: how about `return self.to_nntr_weights(tensorOrList)` ?

File: test/unittest/unittest_nntrainer_models.cpp [hunk 3]
```diff
@@ -928,6 +993,11 @@ INSTANTIATE_TEST_CASE_P(
       mkModelIniTc(gru_return_sequence_with_batch, "2:1:2:1", 10, ModelTestOption::ALL),
       mkModelIniTc(multi_gru_return_sequence, "1:1:1:1", 10, ModelTestOption::ALL),
       mkModelIniTc(multi_gru_return_sequence_with_batch, "2:1:1:1", 10, ModelTestOption::ALL),
+      mkModelIniTc(gru_reset_after_basic, "1:1:1:1", 10, ModelTestOption::ALL),
```
Comment (zhoonit) [inline test/unittest/unittest_nntrainer_models.cpp:996]: Could we test for bigger dimension?

File: nntrainer/layers/grucell.cpp (hunk 1 above)
Comment (lhs8928) [reply nntrainer/layers/grucell.cpp]: Right thanks

File: test/input_gen/transLayer.py (hunk 2 above)
Comment (lhs8928) [reply test/input_gen/transLayer.py]: 👍

File: test/unittest/unittest_nntrainer_models.cpp (hunk 3 above)
Comment (lhs8928) [reply test/unittest/unittest_nntrainer_models.cpp:996]: Done

**Review by zhoonit — APPROVED** [review]
LGTM, nice one :)

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**